packages/shared/src/write-archive/index.ts

```diff
@@ -56,12 +56,23 @@ export async function writeTestResult(
 
       const archiveFile = new ArchiveFile(url, response, pageUrl);
       const origSrcPath = archiveFile.originalSrc();
-      const fileSystemPath = archiveFile.toFileSystemPath();
+      let fileSystemPath = archiveFile.toFileSystemPath();
 
       if (origSrcPath !== fileSystemPath) {
         sourceMap.set(origSrcPath, fileSystemPath);
       }
 
+      // detect the /@fs/ prefix and convert to a relative path that we can add to the archive dir
+      if (fileSystemPath.startsWith('/@fs/')) {
+        const fsPathSlug = fileSystemPath.split('@fs/')[1];
+
+        const overlap = findPathOverlap(fsPathSlug, archiveDir);
+        // with an overlap over a certain length combined with the @fs finding, we can assume a repeated system path
+        if (overlap.length > 30) {
+          fileSystemPath = fsPathSlug.split(overlap)[1];
+        }
```

Comment:
(Requested change) Thinking about this more, we want this take-only-the-actual-matched-path stuff to happen before we call `archiveFile.toFileSystemPath()`. That way we're matching on raw paths before we mess with things (to sanitize or normalize the path). And that way the `sourceMap` has the accurate `fileSystemPath` value that we'll be using.